tools/cli/commands-packages.js

```diff
@@ -1686,11 +1705,13 @@ main.registerCommand({
 
       // If the user explicitly mentioned some packages to upgrade, they must
       // actually end up in our solution!
-      _.each(options.args, function (packageName) {
-        if (! projectContext.packageMap.getInfo(packageName)) {
-          buildmessage.error(packageName + ': package is not in the project');
-        }
-      });
+      if (options.args.length !== 0) {
```

Comment:
Ok.  Seems okay since with your conditional since `upgradePackageNames` should only be non-Cordova packages that they specified on the command line.

Also, I would probably just write this as:

    if (options.args.length) {

...since it's an array and the only truth-y value possible is non-zero.